ThingIFSDK/ThingIFSDK/QueryClause.swift

```diff
@@ -8,6 +8,491 @@
+        }
+    }
+
+    /** Get QueryClause as [ String : Any ] instance
+
+    - Returns: a [ String : Any ] instance.
+    */
+    open func makeDictionary() -> [ String : Any ] {
+        var retval = [
+          "type" : "range",
+          "field" : self.field
+        ] as [String : Any]
+
+        if let lower = self.lower {
+            retval["lowerLimit"] = lower.limit as Any
+            retval["lowerIncluded"] = lower.included
+        }
+        if let upper = self.upper {
+            retval["upperLimit"] = upper.limit as Any
+            retval["upperIncluded"] = upper.included
+        }
+        return retval
+    }
+}
+
+/** Class represents And clause. */
+open class AndQueryClause: NSObject, QueryClause {
+    /** clauses array of AndQueryClause */
+    open private(set) var clauses = [QueryClause]()
+
+    /** Initialize with clause clauses.
+
+    - Parameter clauses: QueryClause instances for AND clauses
+    */
+    public convenience init(_ clauses: QueryClause...) {
+        self.init(clauses)
+    }
+
+    /** Initialize with clause clauses.
+
+     - Parameter clauses: QueryClause array for AND clauses
+     */
+    public init(_ clauses: [QueryClause]) {
+        self.clauses = clauses
+        super.init()
+    }
+
+    public required convenience init(coder aDecoder: NSCoder) {
+        self.init(aDecoder.decodeObject() as! [QueryClause])
+    }
+
+    open func encode(with aCoder: NSCoder) {
+        aCoder.encode(self.clauses)
+    }
+
+    /** Add clause to AndQueryClause
+
+    - Parameter clause: QueryClause instances to add
+    */
+    open func add(_ clause: QueryClause) {
+        self.clauses.append(clause)
+    }
+
+    /** Get QueryClause as [ String : Any ] instance
+
+    - Returns: a [ String : Any ] instance.
+    */
+    open func makeDictionary() -> [ String : Any ] {
+        var array: [[String : Any]] = []
+        self.clauses.forEach { clause in array.append(clause.makeDictionary()) }
+        return ["type": "and", "clauses": array] as [ String : Any ]
+    }
 }
+
+/** Class represents Or clause. */
+open class OrQueryClause: NSObject, QueryClause {
+    /** clauses array of OrQueryClause */
+    open private(set) var clauses = [QueryClause]()
+
+    /** Initialize with clause clauses.
+
+     - Parameter clauses: QueryClause array for OR clauses
+     */
+    public init(_ clauses: [QueryClause]) {
+        self.clauses = clauses
+        super.init()
+    }
+
+    /** Initialize with clause clauses.
+
+    - Parameter clauses: QueryClause instances for OR clauses
+    */
+    public convenience init(_ clauses: QueryClause...) {
+        self.init(clauses)
+    }
+
+    public required convenience init(coder aDecoder: NSCoder) {
+        self.init(aDecoder.decodeObject() as! [QueryClause])
+    }
+
+    open func encode(with aCoder: NSCoder) {
+        aCoder.encode(self.clauses)
+    }
+
+    /** Add clause to OrQueryClause
+
+    - Parameter clause: QueryClause instances to add
```

Comment:
@yongpingchen I think this should be QueryClause because this is a parameter to be passed to Or clause. The parameter can be any of clauses.